geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert/SimpleFeatureConverterFactory.scala

```diff
@@ -234,8 +315,16 @@ trait ToSimpleFeatureConverter[I] extends SimpleFeatureConverter[I] with LazyLog
     new EvaluationContextImpl(names, values, counter)
   }
 
-  override def processInput(is: Iterator[I], ec: EvaluationContext): Iterator[SimpleFeature] =
-    is.flatMap(i =>  processSingleInput(i, ec))
+  override def processInput(is: Iterator[I], ec: EvaluationContext): Iterator[SimpleFeature] = {
+    parseOpts.parseMode match {
+       case ParseMode.Incremental  =>
+         is.flatMap(i => processSingleInput(i, ec))
+       case ParseMode.Batch =>
```

Comment:
(not necessarily for this pr but) for ingestion it could be useful to hook batch size into a flush of the feature writer - there were some requests recently to know which records were actually ingested - this would be a way to do that. (of course, the pre-validation mitigates that to a large extent)